### PR TITLE
[DSS-274] Customize storybook

### PR DESCRIPTION
**Pull request by QuintonJason**
# Description
The intent of this PR is to customize our existing storybook implementation with Sage branding 

- [x] Added a theme to our current storybook implement that is approved by design
- [x] Update structure of the sidebar to be consistent in format and additional sections
- [ ] ~add dark mode theme~ [DSS-306](https://kajabi.atlassian.net/browse/DSS-306)
- [ ] ~add copy for Design contribution~ [DSS-337](https://kajabi.atlassian.net/browse/DSS-337)

**Dependancies:**
-  [ ] ~`storybook-dark-mode`~ [DSS-306](https://kajabi.atlassian.net/browse/DSS-306)

Fixes #(issue)
[DSS-274](https://kajabi.atlassian.net/browse/DSS-274)

Screen Shot 2023-02-10 at 10 15 02 AM

## Type of change
- [x] This change requires a documentation update

# How Has This Been Tested?
docs only no tests needed for this work

**Test Configuration**:
n/a
- sage versions:
- os:
- browsers:
- screen readers:
- misc:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes

[DSS-274]: https://kajabi.atlassian.net/browse/DSS-274?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DSS-306]: https://kajabi.atlassian.net/browse/DSS-306?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DSS-306]: https://kajabi.atlassian.net/browse/DSS-306?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DSS-337]: https://kajabi.atlassian.net/browse/DSS-337?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/core/.storybook/SageTheme.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  barBg: 'white',
+
+  // Form colors
+  inputBg: 'white',
+  inputBorder: '#4B9611',
+  inputTextColor: 'black',
+  inputBorderRadius: 6,
+
+  brandTitle: 'My custom storybook',
+  brandUrl: 'https://example.com',
```
Comment (ju-Skinner) [inline libs/core/.storybook/SageTheme.js]: do we need this here? If so, is this the correct URL we want to use.

File: libs/core/.storybook/preview.js [hunk 2]
```diff
@@ -25,5 +25,21 @@ export const parameters = {
...
+        'Resources',
+        'Foundations',
+        [
+          'Design Principles',
+          'Colors',
+          'Typography'
+        ],
+        'Components',
+        'Get Support'
+      ],
```
Comment (ju-Skinner) [inline libs/core/.storybook/preview.js]: ℹ️  This assists with creating an order of the MDX files in the left Sidebar. 

Do you foresee this as overhead should we want to change the order?

File: libs/core/assets/styles/storybook.css [hunk 3]
```diff
@@ -0,0 +1,5 @@
+/* .docs-hero {
+  background-image: url("/images/hero-background.svg");
+  background-size: cover;
+  padding: 12.5rem 1.5rem 14.25rem;
+} */
```
Comment (ju-Skinner) [inline libs/core/assets/styles/storybook.css]: Do we need this file?

File: libs/core/src/components/my-component/stories/my-component.stories.mdx [hunk 4]
```diff
@@ -3,7 +3,7 @@ import { html, render } from 'lit-html';
 
 import { extractArgTypes } from '@pxtrn/storybook-addon-docs-stencil';
 
-<Meta title="Example/Atoms/my-component" component="my-component" argTypes={extractArgTypes('my-component')} />
+<Meta title="Patterns/my-component" component="my-component" argTypes={extractArgTypes('my-component')} />
```
Comment (ju-Skinner) [inline libs/core/src/components/my-component/stories/my-component.stories.mdx]: I think it's ok to delete this whole file.

File: libs/core/src/stories/Intro/index.js [hunk 5]
```diff
@@ -0,0 +1 @@
+// export default from './Intro';
```
Comment (ju-Skinner) [inline libs/core/src/stories/Intro/index.js]: This isn't being used any longer.

File: libs/core/src/stories/Intro/Welcome.stories.mdx [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import { Meta, Story } from '@storybook/addon-docs';
+import Intro from './Intro'
```
Comment (ju-Skinner) [inline libs/core/src/stories/Intro/Welcome.stories.mdx]: This doesn't seem to be used

**Review by ju-Skinner — CHANGES_REQUESTED** [review]
see comments below

**Comment by goodwinchris** [comment]
This looks awesome!

I'm just seeing a few little alignment issues in the sidebar, looks like the change in font-family broke the alignment between text and the icon a bit:
Before|After
--|--
CleanShot 2023-02-14 at 11 06 42@2x|CleanShot 2023-02-14 at 16 22 54@2x

Also seeing the old hover color, I wonder if we can get the hover to match the Sage leaf color?
CleanShot 2023-02-14 at 11 15 04@2x

File: libs/core/.storybook/preview.js (hunk 2 above)
Comment (QuintonJason) [reply libs/core/.storybook/preview.js]: > This looks awesome!
> 
> I'm just seeing a few little alignment issues in the sidebar, looks like the change in font-family broke the alignment between text and the icon a bit:
> 
> Before	After
> CleanShot 2023-02-14 at 11 06 42@2x	CleanShot 2023-02-14 at 16 22 54@2x

Unfortunately we don't have a way to update the positioning of the icons of text. There are a couple storybooks that change the font and they all have this problem.

> Also seeing the old hover color, I wonder if we can get the hover to match the Sage leaf color? CleanShot 2023-02-14 at 11 15 04@2x

Yep, I'll remove the added font

**Comment by QuintonJason** [comment]
> This looks awesome!
> 
> I'm just seeing a few little alignment issues in the sidebar, looks like the change in font-family broke the alignment between text and the icon a bit:
> 
> Before	After
> CleanShot 2023-02-14 at 11 06 42@2x	CleanShot 2023-02-14 at 16 22 54@2x
> Also seeing the old hover color, I wonder if we can get the hover to match the Sage leaf color? CleanShot 2023-02-14 at 11 15 04@2x

- Can remove the font update
- The hover color is consistent like that across all the customized storybooks. It's not one of the options

File: libs/core/src/stories/Contributing.stories.mdx [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+### Meetings and events [TBD]
+We are exploring offering new types of meetings for the community to bring questions, comments, ideas, and feedback:
+
+**Office Hours — Every Tue/Thur [TBD]**
+- **Purpose**
+For product designers seeking feedback on their usage of design systems and visual language. This is also a great time to propose new components or component changes.
+- **Audience**
+Primarily designers, though engineers and product managers are welcome if there are technical or strategic concerns as well.
+- **Cadence**
+Twice a week on Tuesdays and Thursdays. Please sign up for a half-hour slot with your discussion topic.
```
Comment (ju-Skinner) [inline libs/core/src/stories/Contributing.stories.mdx]: Lets pull this out...This feels more like for internal use only.

**Review by ju-Skinner — CHANGES_REQUESTED** [review]
See comment regarding `Contributing.stories.mdx`. Other than that everything else looks great. Great effort and work. 👏🏾

File: libs/core/src/stories/Contributing.stories.mdx [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+
+## Resources, Slack, and meetings
+### Jira and confluence
+If you’re curious what we’re working on, you can check out our [Sage Design Board](https://kajabi.atlassian.net/jira/software/c/projects/DS/boards/205) or the [Design Systems Services Board](https://kajabi.atlassian.net/jira/software/projects/DSS/boards/216).
+
+To see the bigger picture, you can view [our roadmap](https://kajabi.atlassian.net/jira/software/projects/DSS/boards/216/roadmap) and priorities for each quarter. These priorities are determined through partnership meetings, which we use to learn about component needs and coordinate with designers to determine timelines.
+
+### Slack channels
+Before reaching out, take a look at [our documentation](https://sage-design-system.kajabi.com/pages/index) to see if it answers your question, because it will likely get you the fastest answer. Still need help? Try searching Slack for your question, and then feel free to ask if your question hasn’t been answered in the past.
```
Comment (pixelflips) [inline libs/core/src/stories/Contributing.stories.mdx]: the old URL for documentation here: https://sage.kajabi.com is correct now.

**Review by goodwinchris — APPROVED** [review]

File: libs/core/src/stories/Welcome.stories.mdx [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+
+## Foundations
+Our system starts with some core visual elements: Type, Color, and Icons. We also prescribe common practices for aspects of the system.
+
+## Components
+Components are the backbone of our design system. In order to be most usable, our components are written in both Rails and React form.
+Code examples, property options, and best practices can all be discovered here. Links to Figma design documentation can also be found in this section.
+
+## Contributing
+We encourage any member of the Kajabi team to feel empowered to participate. Understanding and contributing to the system are the keys to the system's longevity.
```
Comment (ju-Skinner) [inline libs/core/src/stories/Welcome.stories.mdx]: Are we only building this for Kajabi? If so, lets make the repo private

cc: @monicawheeler

File: libs/core/src/stories/Welcome.stories.mdx [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+<Meta title="Welcome" parameters={{ previewTabs: { canvas: { hidden: true }, } }} />
+
+# Building with Sage
+Our system is our source of truth, providing everything you need to build the best experiences for all Kajabi entrepreneurs.
+
+## Foundations
+Our system starts with some core visual elements: Type, Color, and Icons. We also prescribe common practices for aspects of the system.
+
+## Components
+Components are the backbone of our design system. In order to be most usable, our components are written in both Rails and React form.
```
Comment (ju-Skinner) [inline libs/core/src/stories/Welcome.stories.mdx:12]: >Components are the backbone of our design system. In order to be most usable, our components are written in both Rails and React form.

This is not the case...we are building one code base using Web Components

File: libs/core/src/stories/Contributing.stories.mdx (hunk 7 above)
Comment (ju-Skinner) [reply libs/core/src/stories/Contributing.stories.mdx]: Should we remove this as it feels like this is meant and targeted for Internal Use only?

cc: @monicawheeler

**Review by ju-Skinner — CHANGES_REQUESTED** [review]
Found some additional areas that should be addressed. Also included @monicawheeler to get her thoughts as well.

File: libs/core/src/stories/Welcome.stories.mdx (hunk 9 above)
Comment (pixelflips) [reply libs/core/src/stories/Welcome.stories.mdx]: So far! :nitpick 🤣

**Review by pixelflips — APPROVED** [review]
LGTM! Great work!

File: libs/core/src/stories/Welcome.stories.mdx (hunk 9 above)
Comment (monicawheeler) [reply libs/core/src/stories/Welcome.stories.mdx]: We want our Core Design System repo (this repo) to be public facing, so our language in documentation should be geared that way.

```suggestion
We encourage everyone to feel empowered to participate. Understanding and contributing to the system are the keys to the system's longevity.
```

File: libs/core/src/stories/Contributing.stories.mdx [hunk 11]
```diff
@@ -0,0 +1,89 @@
+
+import { Meta, Story } from '@storybook/addon-docs';
+
+<Meta title="Resources/Contributing to Sage"  parameters={{ previewTabs: { canvas: { hidden: true }, } }} />
+
+# How to work with Sage
+Guidelines on how to engage with the Sage team, when to work with us, and how to contribute.
```
Comment (monicawheeler) [inline libs/core/src/stories/Contributing.stories.mdx]:
```suggestion
Guidelines on engaging with the Design System team, when to work with us, and how to contribute.
```

File: libs/core/src/stories/Contributing.stories.mdx [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+
+**Team Charters**
```
Comment (monicawheeler) [inline libs/core/src/stories/Contributing.stories.mdx]: Private documentation - let's remove this.

File: libs/core/src/stories/Contributing.stories.mdx [hunk 13]
```diff
@@ -0,0 +1,89 @@
+
```
Comment (monicawheeler) [inline libs/core/src/stories/Contributing.stories.mdx:1]: Let's rework this document to be more development focused for now.

**Review by ju-Skinner — APPROVED** [review]

**Pull request merged**